malaria-ios/ViewControllers/MythVsFactViewController.swift

```diff
@@ -0,0 +1,83 @@
+import UIKit
+
+class MythVsFactViewController: GameViewController {
+  
+  @IBOutlet weak var statementNumberLabel: UILabel!
+  @IBOutlet weak var statementLabel: UILabel!
+  
+  @IBOutlet weak var yesView: UIImageView!
+  @IBOutlet weak var noView: UIImageView!
+  
+  var MVFGame: MythVsFactGame!
+  
+  var statementLabelInitialFrame: CGRect!
+  
+  override var currentLevel: Int {
+    didSet {
+      
+      // Check if we ran out of questions
+      if currentLevel >= MVFGame!.entries.count {
+        endGame()
+        return
+      }
+      
+      // Set the statement
+      statementLabel.text = MVFGame?.entries[currentLevel].statement
+      statementNumberLabel.text = "Statement \(currentLevel + 1) / \(MVFGame!.entries.count)"
+    }
+  }
+  
+  override func viewWillAppear(animated: Bool) {
+    
+    MVFGame = game as! MythVsFactGame
```

Comment:
Instead of doing this (repeatedly in every game)

You can try to use generics. I.e.

```swift
class GameViewController<T> where T : Game{
/// Presents game rules.
class GameRulesViewController : UIViewController {
  
  @IBOutlet weak var gameTitleLabel: UILabel!
  @IBOutlet weak var gameDescriptionLabel: UILabel!
  
  // Provided by previous viewController
  var game: T!
}
```

Then:
```swift
class MyViewController : GameViewController<RapidFireGame> {

}
```

Would clean up the code (no need for this viewWillAppear or similar) and simply further extensions ;) Moreover it would having making casts
Check if it is feasible.